modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1126,6 +1128,196 @@ public void testEnableAccountEntryRestricted() throws Exception {
 		}
 	}
 
+	@Test
+	public void testEnableAccountEntryRestrictedForNondefaultStorageType()
+		throws Exception {
+
+		// Enabling account restriction to a custom salesforce object definition
+
+		ObjectDefinition objectDefinition1 =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false, false,
+				LocalizedMapUtil.getLocalizedMap("Able"), "Able", null, null,
+				LocalizedMapUtil.getLocalizedMap("Ables"),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+				Collections.emptyList());
+
+		ObjectField objectField1 = ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				TestPropsValues.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				StringUtil.randomId()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).required(
+				true
+			).build());
+
+		objectDefinition1 =
+			_objectDefinitionLocalService.
+				enableAccountEntryRestrictedForNondefaultStorageType(
+					objectField1);
+
+		Assert.assertTrue(objectDefinition1.isAccountEntryRestricted());
+		Assert.assertTrue(
+			objectDefinition1.getAccountEntryRestrictedObjectFieldId() > 0);
+		Assert.assertFalse(objectDefinition1.isSystem());
+		Assert.assertEquals(
+			objectDefinition1.getStorageType(),
+			ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE);
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition1);
+
+		// Switching account restricted field
+
+		ObjectDefinition objectDefinition2 =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false, false,
+				LocalizedMapUtil.getLocalizedMap("Able"), "Able", null, null,
+				LocalizedMapUtil.getLocalizedMap("Ables"),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+				Collections.emptyList());
+
+		ObjectField objectField2 = ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				TestPropsValues.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				StringUtil.randomId()
+			).objectDefinitionId(
+				objectDefinition2.getObjectDefinitionId()
+			).required(
+				true
+			).build());
+
+		ObjectField objectField3 = ObjectFieldUtil.addCustomObjectField(
```

Comment:
Declare the variable next to its usage, please.

Reply:
After moving this close to its usage, maybe we could reuse the variable objectField instead of creating a new one.